src/repo/fs/blocks.rs

```diff
@@ -100,6 +100,57 @@ impl FsBlockStore {
             }
         }
     }
+
+    /// Implementation moved here from inside the `impl BlockStore for FsBlockStore` which is
+    /// `#[async_trait]`, and somehow the two lifetimes generated by `#[async_trait]` just don't
+    /// work leading to confusing error when `ipfs` is used as a dependency in an outside of
+    /// workspace crate.
+    ///
+    /// See [gist] for the standalone working workaround, error and an expanded version of the
+    /// original code.
+    ///
+    /// [gist]: https://gist.github.com/koivunej/d6abccb4133839eeab8b36992f1a95fa
+    async fn list0(&self) -> Result<Vec<Cid>, Error> {
```

Comment:
If you wanted, `list0` can be nested inside `list` (swapping `&self` for `myself: &FsBlockStore`) without bringing back the compilation error to keep things more organized.

Reply:
Agreed, good idea. Thanks for taking a peek!

Reply:
Baked this change in the first commit, resolving..